Amplify/Categories/API/Operation/RetryableGraphQLOperation.swift

```diff
@@ -6,183 +6,150 @@
 //
 
 import Foundation
+import Combine
 
-/// Convenience protocol to handle any kind of GraphQLOperation
-public protocol AnyGraphQLOperation {
-    associatedtype Success
-    associatedtype Failure: Error
-    typealias ResultListener = (Result<Success, Failure>) -> Void
-}
-
-/// Abastraction for a retryable GraphQLOperation.
-public protocol RetryableGraphQLOperationBehavior: Operation, DefaultLogger {
-    associatedtype Payload: Decodable
-
-    /// GraphQLOperation concrete type
-    associatedtype OperationType: AnyGraphQLOperation
-
-    typealias RequestFactory = () async -> GraphQLRequest<Payload>
-    typealias OperationFactory = (GraphQLRequest<Payload>, @escaping OperationResultListener) -> OperationType
-    typealias OperationResultListener = OperationType.ResultListener
-
-    /// Operation unique identifier
-    var id: UUID { get }
-
-    /// Number of attempts (min 1)
-    var attempts: Int { get set }
-
-    /// Underlying GraphQL operation instantiated by `operationFactory`
-    var underlyingOperation: AtomicValue<OperationType?> { get set }
 
-    /// Maximum number of allowed retries
-    var maxRetries: Int { get }
-
-    /// GraphQLRequest factory, invoked to create a new operation
-    var requestFactory: RequestFactory { get }
-
-    /// GraphQL operation factory, invoked with a newly created GraphQL request
-    /// and a wrapped result listener.
-    var operationFactory: OperationFactory { get }
+// MARK: - RetryableGraphQLOperation
+public final class RetryableGraphQLOperation<Payload: Decodable> {
```

Comment:
same for this, would be great to see more generic Amplify-level unit tests against this class (if it doesn't exist already)

Reply:
The [unit test cases](https://github.com/aws-amplify/amplify-swift/pull/3666/files#diff-27ea1d73052a74a6ba5d49e97d7d53a779f3e5c6e0a32217f1d5e49ed4abac45R10) of this class has been updated and collapsed by the github diff tool.